lib/src/main/java/graphql/nadel/engine/blueprint/hydration/NadelHydrationWhenCondition.kt

```diff
@@ -0,0 +1,34 @@
+package graphql.nadel.engine.blueprint.hydration
+
+import graphql.nadel.engine.transform.query.NadelQueryPath
+
+sealed class NadelHydrationWhenCondition {
+    abstract fun evaluate(resultId: String): Boolean
```

Comment:
`resultId` should be `Any` (maybe nullable too) probably. But we can fix that up in another PR.